helper-scripts/backup_and_restore.sh

```diff
@@ -1,331 +1,672 @@
-    echo "${BACKUP_LOCATION} is not write-able for others, that's required for a backup."
+}
+
+function check_valid_restore_directory() {
+  RESTORE_LOCATION="${1}"
+
+  if [[ -z "${RESTORE_LOCATION}" ]]; then
+    echo
+    echo -e "${RED_COLOR}Fatal Error: restore location not specified!${RESET}"
+    exit 1
+  fi
+
+  if [[ ! -e "${RESTORE_LOCATION}" ]]; then
+    echo -e "${RED_COLOR}${RESTORE_LOCATION} is not exist${RESET}"
+    exit 1
+  fi
+
+  if [[ ! -d "${RESTORE_LOCATION}" ]]; then
+    echo -e "${RED_COLOR}${RESTORE_LOCATION} is not a valid path! Maybe a file or a symbolic?${RESET}"
     exit 1
   fi
+}
+
+function delete_old_backups() {
+  if [[ -z $(find "${MAILCOW_DELETE_LOCATION}"/mailcow-* -maxdepth 0 -mmin +$((${MAILCOW_DELETE_DAYS}*60*24)) 2> /dev/null) ]]; then
+    echo -e "${YELLOW_COLOR}No backups to delete found.${RESET}"
+    exit 0
+  fi
+
+  echo -e "${BLUE_COLOR}Backups scheduled for deletion:${RESET}"
+  find "${MAILCOW_DELETE_LOCATION}"/mailcow-* -maxdepth 0 -mmin +$((${MAILCOW_DELETE_DAYS}*60*24)) -exec printf "${YELLOW_COLOR}- %s${RESET}\n" {} \;
+
+  if [[ "${MAILCOW_YES_TO_ALL}" = "1" ]]; then
+    echo -e "${GREEN_COLOR}Start deleting...${RESET}"
+    find "${MAILCOW_DELETE_LOCATION}"/mailcow-* -maxdepth 0 -mmin +$((${MAILCOW_DELETE_DAYS}*60*24)) -exec rm -rvf {} \;
+    echo -e "${GREEN_COLOR}Done.${RESET}"
+  else
+    read -p "$(echo -e "${BOLD}${RED_COLOR}Are you sure you want to delete the above backups? type YES in capital letters to delete, else to skip: ${RESET}")" DELETE_CONFIRM
+    if [[ "${DELETE_CONFIRM}" == "YES" ]]; then
+      find "${MAILCOW_DELETE_LOCATION}"/mailcow-* -maxdepth 0 -mmin +$((${MAILCOW_DELETE_DAYS}*60*24)) -exec rm -rvf {} \;
+    else
+      echo -e "${YELLOW_COLOR}OK, skipped.${RESET}"
+    fi
+  fi
+}
+
+# ----------------- End Functions -----------------
+
+check_required_tools
+
+DEBIAN_DOCKER_IMAGE="mailcow/backup:latest"
+
+# This duplicated assignment is to address
+# issue in `https://github.com/mailcow/mailcow-dockerized/issues/957`
+if [[ ! -z "${MAILCOW_BACKUP_LOCATION}" ]]; then
+  BACKUP_LOCATION="${MAILCOW_BACKUP_LOCATION}"
 fi
 
-BACKUP_LOCATION=$(echo ${BACKUP_LOCATION} | sed 's#/$##')
+if [[ $# -eq 0 ]]; then
+  print_usage
+  exit 0
+fi
+
+MAILCOW_BACKUP_COMPONENTS=()
+
+# These variables can be set using flags or environment variables
+# Such as `./script.sh -b /path/to/backup`
+# Or `MAILCOW_BACKUP_LOCATION=/path/to/backup ./script.sh`
+MAILCOW_BACKUP_LOCATION="${MAILCOW_BACKUP_LOCATION:-}"
+MAILCOW_RESTORE_LOCATION="${MAILCOW_RESTORE_LOCATION:-}"
+MAILCOW_DELETE_LOCATION="${MAILCOW_DELETE_LOCATION:-}"
+MAILCOW_DELETE_DAYS="${MAILCOW_DELETE_DAYS:-}"
+MAILCOW_BACKUP_RESTORE_THREADS="${MAILCOW_BACKUP_RESTORE_THREADS:-1}"
+MAILCOW_YES_TO_ALL="${MAILCOW_YES_TO_ALL:-0}"
+
+# Check for required files
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 COMPOSE_FILE=${SCRIPT_DIR}/../docker-compose.yml
```

Comment:
can we also accept compose file and env from env if possible, it would be nice to keep all of my compose in a set directory structure.

Reply:
and ofcourse default should be the one, which we already have.

Reply:
> can we also accept compose file and env from env if possible, it would be nice to keep all of my compose in a set directory structure.

Hey @DerLinkman , please check is this possible/should be implemented?

Reply:
No i don't like that. Sorry we have to keep mailcow as it is...